src/stages/normalize/patchers/ClassPatcher.js

```diff
@@ -2,9 +2,17 @@ import { SourceType } from 'coffee-lex';
 
 import NodePatcher from './../../../patchers/NodePatcher';
 import AssignOpPatcher from './AssignOpPatcher';
+import BlockPatcher from './BlockPatcher';
+import ProgramPatcher from './ProgramPatcher';
 
 import type { PatcherContext } from './../../../patchers/types';
 
+const INIT_CLASS_HELPER = `\
+__initClass__ = (c) ->
```

Comment:
Ooh, good idea. Still feels like sort of a hack (e.g. it could potentially mess up static analysis like name deconfliction in the main stage), but seems better than what I have, and of course it fixes the correctness issue you also mentioned.